### PR TITLE
feat: add src type to CloudinaryLoaderCldOptions interface

### LINKED ISSUE #425: [Bug] @ts-ignore in cloudinary-loader.ts
# **Bug Report**

## **Describe the bug**

A @ts-ignore is currently in place in the cloudinaryLoader function which is what organizes and ultimately helps to build a Cloudinary URL based on Next.js props.

https://github.com/cloudinary-community/next-cloudinary/blob/main/next-cloudinary/src/loaders/cloudinary-loader.ts#L58-L59

Goal wouild be to address whatever type issues are preventing this line from being removed.

### PR DESCRIPTION
**Pull request by vickywane**
# Description

This PR will add the `src` type to `CloudinaryLoaderCldOptions` interface to fix the ts error being thrown from the `cloudinary-loader.ts` file.  

Fixes This will close #425 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Fix or improve the documentation
- [ ] This change requires a documentation update

# Checklist

- [x] I have followed the contributing guidelines of this project as mentioned in [CONTRIBUTING.md](/CONTRIBUTING.md)
- [x] I have created an [issue](https://github.com/cloudinary-community/next-cloudinary/issues) ticket for this PR
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/cloudinary-community/next-cloudinary/pulls) for the same update/change?
- [x] I have performed a self-review of my own code
- [x] I have run tests locally to ensure they all pass
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes needed to the documentation

### CODE REVIEW

**Comment by colbyfayock** [comment]
hey @vickywane looks like this is failing

can you try to seei f you can reproduce the error locally by running `pnpm build` inside of the `next-cloudinary` directory?

image

**Comment by colbyfayock** [comment]
@vickywane hey any updates?

**Comment by vickywane** [comment]
@colbyfayock I looked through the code and saw that specifying a `src` property was causing the image URL to be overwritten when spread in [L86 ](https://github.com/cloudinary-community/next-cloudinary/blob/main/next-cloudinary/src/components/CldImage/CldImage.tsx#L86).  

I'm doing a bit of typecasting in [L190](https://github.com/cloudinary-community/next-cloudinary/pull/552/files#diff-879a45f305415a4b663acd8b4463e563c5a05d789e561f405d78d7cd3d5cc707R190) which helps us to still maintain the expected type for the `cloudinaryLoader` function.

**Comment by vickywane** [comment]
@colbyfayock Have you been able to take a look at this?

File: next-cloudinary/src/components/CldImage/CldImage.tsx [hunk 1]
```diff
@@ -79,17 +93,24 @@ const CldImage = forwardRef<HTMLImageElement, CldImageProps>(function CldImage(p
...
-    }, props.config);
+  const IMAGE_OPTIONS: { unoptimized?: boolean } = (process.env
+    .__NEXT_IMAGE_OPTS || {}) as unknown as object;
+
+  if (props.unoptimized === true || IMAGE_OPTIONS?.unoptimized === true) {
+    imageProps.src = getCldImageUrl(
+      {
+        ...cldOptions,
+
+        ...{
```
Comment (colbyfayock) [inline next-cloudinary/src/components/CldImage/CldImage.tsx:104]: why is this needed?

**Comment by colbyfayock** [comment]
i need to find some time to spend reviewing this. its difficult with the formatting changes and i'm not quite sure the fixes are fixing the root cause or a bandaid. will try to find some time this week

**Comment by vickywane** [comment]
@colbyfayock I have removed the formatting to make this easier to review ( sorry that came from my code editor 
:) )

> the fixes are fixing the root cause or a bandaid

I would say this is a fine mix of a bandaid and a fix.

**Comment by colbyfayock** [comment]
thanks. im at a conference but will be back thursday and will try to review

**Comment by vickywane** [comment]
@colbyfayock Okay. 

Looking forward to it!

**Comment by colbyfayock** [comment]
hey @vickywane just merged in the `beta` branch from some work being done on refactoring some of the code, made a few of the changes moot, but still fixes the loader tweaks

thanks for fixing this all. going to merge it back into the `beta` branch to go out sometime soon from there

thanks for your contribution. this PR is eligible for free Cloudinary Hackatoberfest swag. Please send me an email at [hacktoberfest@cloudinary.com](mailto:hacktoberfest@cloudinary.com) with your name, GitHub username, and a link to the PR where I'll send further instructions.

Happy Hacktoberfest!

**Comment by colbyfayock** [comment]
@all-contributors please add @vickywane for code
